repose-aggregator/core/repose-core/src/main/java/org/openrepose/powerfilter/PowerFilter.java

```diff
@@ -469,35 +453,19 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
             }
         } catch (InvalidMethodException ime) {
             LOG.debug("{}:{} -- Invalid HTTP method requested: {}", clusterId, nodeId, wrappedRequest.getMethod(), ime);
-            if (scope != null) {
-                scope.span().setTag(Tags.HTTP_STATUS.getKey(), HttpServletResponse.SC_BAD_REQUEST);
-                scope.span().finish();
-                scope.close();
-            }
+            closeScope(scope, HttpServletResponse.SC_BAD_REQUEST);
             wrappedResponse.sendError(HttpServletResponse.SC_BAD_REQUEST, "Error processing request");
         } catch (URISyntaxException use) {
             LOG.debug("{}:{} -- Invalid URI requested: {}", clusterId, nodeId, wrappedRequest.getRequestURI(), use);
-            if (scope != null) {
-                scope.span().setTag(Tags.HTTP_STATUS.getKey(), HttpServletResponse.SC_BAD_REQUEST);
-                scope.span().finish();
-                scope.close();
-            }
+            closeScope(scope, HttpServletResponse.SC_BAD_REQUEST);
             wrappedResponse.sendError(HttpServletResponse.SC_BAD_REQUEST, "Error processing request");
         } catch (Exception ex) {
             LOG.error("{}:{} -- Issue encountered while processing filter chain.", clusterId, nodeId, ex);
-            if (scope != null) {
-                scope.span().setTag(Tags.HTTP_STATUS.getKey(), HttpServletResponse.SC_BAD_GATEWAY);
-                scope.span().finish();
-                scope.close();
-            }
+            closeScope(scope, HttpServletResponse.SC_BAD_GATEWAY);
             wrappedResponse.sendError(HttpServletResponse.SC_BAD_GATEWAY, "Error processing request");
         } catch (Error e) {
             LOG.error("{}:{} -- Error encountered while processing filter chain.", clusterId, nodeId, e);
-            if (scope != null) {
-                scope.span().setTag(Tags.HTTP_STATUS.getKey(), HttpServletResponse.SC_BAD_GATEWAY);
-                scope.span().finish();
-                scope.close();
-            }
+            closeScope(scope, HttpServletResponse.SC_BAD_GATEWAY);
```

Comment:
Do we need all these `closeScope` calls? Can we just handle it in the `finally` block? You should be able to get the status code even if the response is "committed", and we could repeat ourselves less. Also, repeating the status code in the `closeScope` and `sendError` opens the path for the status to mismatch at some point. Why do that rather than always using the actual status set on the response?

Reply:
Actually, we can probably follow the recommendation of the Java implementation of OpenTracing, minus the try-with-resource construct, in the README here:
https://github.com/opentracing/opentracing-java

That is, I propose:
```
io.opentracing.Tracer tracer = ...;
...
Scope scope = tracer.buildSpan(...).startActive(true);
// asChildOf stuff
try {
    // Do things.
} catch(Exception ex) {
    Tags.ERROR.set(scope.span(), true);
} finally {
    Tags.HTTP_STATUS.set(scope.span(), wrappedResponse.getStatus());
    scope.close();
}
```

Reply:
Yep. I agree. This was the first pass at cleaning up the prototype code that had those three lines copy/pasted all over the place. Again, consistency even within the project is the best practice.